src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.VariableIdentifier.cs

```diff
@@ -24,28 +25,58 @@ public VariableIdentifier(Symbol symbol, int containingSlot = 0)
 
             public bool Exists
             {
-                get { return (object)Symbol != null; }
+                get { Debug.Assert((object)Symbol != null);  return (object)Symbol != null; }
             }
+
             public override int GetHashCode()
             {
-                return Roslyn.Utilities.Hash.Combine(Symbol?.OriginalDefinition, ContainingSlot);
+                Debug.Assert((object)Symbol != null);
+
+                int currentKey = ContainingSlot;
+                int thisIndex = Symbol.MemberIndex;
```

Comment:
>thisIndex [](start = 20, length = 9)

It may be good to leave a comment (why are we using the index whenever one is available?)